rio_tiler/io/base.py

```diff
@@ -34,28 +36,31 @@ class SpatialMixin:
     """
 
     tms: TileMatrixSet = attr.ib(default=WEB_MERCATOR_TMS)
-    bounds: BBox = attr.ib(init=False)
     minzoom: int = attr.ib(init=False)
     maxzoom: int = attr.ib(init=False)
 
-    @property
-    def center(self) -> Tuple[float, float, int]:
-        """Dataset center + minzoom."""
-        return (
-            (self.bounds[0] + self.bounds[2]) / 2,
-            (self.bounds[1] + self.bounds[3]) / 2,
-            self.minzoom,
-        )
+    dataset_bounds: BBox = attr.ib(init=False)
+    dataset_crs: CRS = attr.ib(init=False)
 
     @property
-    def spatial_info(self) -> SpatialInfo:
-        """Return Dataset's spatial info."""
-        return SpatialInfo(
-            bounds=self.bounds,
-            center=self.center,
-            minzoom=self.minzoom,
-            maxzoom=self.maxzoom,
-        )
+    def bounds(self) -> BBox:
+        """Dataset bounds in WGS84."""
+        try:
+            if self.dataset_crs == WGS84_CRS:
+                return self.dataset_bounds
+
+            return transform_bounds(
+                self.dataset_crs, WGS84_CRS, *self.dataset_bounds, densify_pts=21
+            )
+
+        except:  # noqa
+            # If we can't calculate the transformation between dataset CRS and WGS84
+            # bounds default to -180.0, -90.0, 180.0, 90.0
+            warnings.warn(
+                "Cannot dertermine bounds in WGS84 crs, will default to (-180.0, -90.0, 180.0, 90.0).",
+                UserWarning,
+            )
+            return (-180.0, -90.0, 180.0, 90.0)
```

Comment:
we will always default to (-180.0, -90.0, 180.0, 90.0) even if `dataset_bounds` or `dataset_crs` are not set 🤷‍♂️